### PR TITLE
[Translation] [russian] - translates slices into Russian

### PR DESCRIPTION
**Pull request by koleaby4**

### CODE REVIEW

File: _content/tour/rus/slices.article [hunk 1]
```diff
@@ -141,83 +142,79 @@ Output:
...
+а затем возвращает эту копию вызывающей стороне.
 
-Why is the API designed this way? This is because the idiom is to use value semantics
-to move a slice value around a program. This must still be respected even with a
-mutation operation. Plus, value semantic mutation is the safest way to perform mutation
-since the mutation is being performed on the function’s own copy of the data in isolation.
+Почему API разработан таким образом? Это потому, что идиома состоит в использовании семантики
+значений для передвижения значения среза внутри программы. Это необходимо иметь в виду
+даже при операции мутации. Кроме того, семантика мутации значений является самым безопасным
+способом выполнения мутации, поскольку мутация выполняется на собственной копии данных в рамках функции.
```
Comment (rumyantseva) [inline _content/tour/rus/slices.article:168]:
```suggestion
Почему API разработан таким образом? Дело в том, что идиоматичный подход состоит в использовании семантики
значений для передачи значения среза внутри программы. Это необходимо иметь в виду
даже при операции мутации. Кроме того, мутация на основе семантики значений является самым безопасным
способом выполнения мутации, поскольку мутация выполняется на собственной копии данных функции в изоляции.
```

File: _content/tour/rus/slices.article [hunk 2]
```diff
@@ -141,83 +142,79 @@ Output:
...
 
-** Slicing Slices
+** Создание срезов из срезов
 
-Slices provide the ability to avoid extra copies and heap allocations of the backing
-array when needing to isolate certain elements of the backing array for different
-operations.
+Срезы дают возможность избежать дополнительного копирования и выделения памяти
+для основного массива,  когда нужно выделить определенные элементы основного массива
+для различных операций.
```
Comment (rumyantseva) [inline _content/tour/rus/slices.article:199]:
```suggestion
Срезы дают возможность избежать дополнительного копирования и выделения памяти на куче 
для базового массива,  когда нужно выделить определенные элементы этого базового массива
для различных операций.
```

File: _content/tour/rus/slices.article [hunk 3]
```diff
@@ -141,83 +142,79 @@ Output:
...
 
     slice1 := []string{"A", "B", "C", "D", "E"}
     slice2 := slice1[2:4]
 
-The variable slice2 is a new slice value that is now sharing the same backing
-array that slice1 is using. However, slice2 only allows you to access the elements
-at index 2 and 3 (C and D) of the original slice’s backing array. The length of
-slice2 is 2 and not 5 like in slice1 and the capacity is 3 since there are now
-3 elements from that pointer position.
+Переменная slice2 - это новое значение среза, которое сейчас разделяет тот же исходный
```
Comment (rumyantseva) [inline _content/tour/rus/slices.article:207]:
```suggestion
Переменная slice2 - это новое значение среза, которое сейчас разделяет тот же базовый
```

File: _content/tour/rus/slices.article [hunk 4]
```diff
@@ -141,83 +142,79 @@ Output:
...
+массив, что и slice1. Однако slice2 позволяет обращаться только к элементам с индексами
+2 и 3 (C и D) исходного массива среза. Длина slice2 равна 2, а не 5, как в slice1, а
```
Comment (rumyantseva) [inline _content/tour/rus/slices.article:209]:
```suggestion
2 и 3 (C и D) исходного базового массива среза. Длина slice2 равна 2, а не 5, как в slice1, а
```

File: _content/tour/rus/slices.article [hunk 5]
```diff
@@ -141,83 +142,79 @@ Output:
...
+ёмкость равна 3, так как сейчас есть 3 элемента с этой позиции указателя.
 
-.image /tour/eng/static/img/sl4.png
+.image /tour/rus/static/img/sl4.png
 
-A better way to think about slicing is to focus on the length using this notation
-[a:a+len]  index a through a plus the length. This will reduce errors in
-calculating new slices.
+Лучше всего думать о срезе, фокусируясь на длине с помощью следующей нотации
+[a:a+len] - индекс `a` плюс длина.
```
Comment (rumyantseva) [inline _content/tour/rus/slices.article:215]:
```suggestion
Лучше всего думать о срезе, фокусируясь на длине с помощью следующей нотации
[a:a+len] - индекс `a` плюс длина. Это уменьшит ошибки при вычислении новых срезов.
```

File: _content/tour/rus/slices.article [hunk 6]
```diff
@@ -245,29 +242,29 @@ Output:
...
 
-The nice thing here is there are no allocations. The compiler knows the size of
-the backing array for slice1 at compile time. Passing a copy of the slice value
-down into the inspectSlice function keeps everything on the stack.
+Здесь хорошо то, что не происходит выделения памяти на куче. Компилятор знает размер
+исходного массива для slice1 на этапе компиляции. Передача копии значения среза
+в функцию inspectSlice позволяет хранить все значения на стеке.
 
-** Mutations To The Backing Array
+** Изменения исходного массива
```
Comment (rumyantseva) [inline _content/tour/rus/slices.article:255]:
```suggestion
** Изменения базового массива
```

File: _content/tour/rus/slices.article [hunk 7]
```diff
@@ -357,18 +353,18 @@ Output:
...
-Notice after the call to append, slice2 has a new backing array.
+Обратите внимание, после вызова append, у slice2 появился новый базовый массив.
 
-.image /tour/eng/static/img/sl7.png
+.image /tour/rus/static/img/sl7.png
 
-This can be seen by comparing the addresses of each slice. In this case, the
-mutation against slice2 didn’t cause a side effect against slice1.
+Это видно при сравнении адресов срезов. В этом случае изменение значений
+в slice2 не вызвала побочного эффекта для slice1.
```
Comment (rumyantseva) [inline _content/tour/rus/slices.article:361]:
```suggestion
в slice2 не вызвало побочного эффекта для slice1.
```

File: _content/tour/rus/slices.article [hunk 8]
```diff
@@ -392,18 +388,18 @@ Output:
...
+** Срезы используют мутацию семантики указателя
 
-It’s important to remember that even though you use value semantics to move a slice
-around the program, when reading and writing a slice, You are using pointer semantics.
-Sharing individual elements of a slice with different parts of my program can
-cause unwanted side effects.
+Важно помнить, что даже если вы используете семантику значения для перемещения
+среза по программе, при чтении и записи среза вы используете семантику указателя.
+Совместное использование отдельных элементов среза различными частями программы может
+привести к нежелательным побочным эффектам.
```
Comment (rumyantseva) [inline _content/tour/rus/slices.article:399]:
```suggestion
Важно помнить, что даже если вы используете семантику значения для передачи
среза по программе, при чтении и записи среза вы используете семантику указателя.
Совместное использование отдельных элементов среза различными частями программы может
привести к нежелательным побочным эффектам.
```

File: _content/tour/rus/slices.article [hunk 9]
```diff
@@ -441,23 +437,22 @@ Output:
...
+и делиться значением структуры.
 
-** Linear Traversal Efficiency
-The beauty of a slice is its ability to allow for performing linear traversals
-that are mechanically sympathetic while sharing data using value semantics to
-minimize heap allocations.
+** Эффективность линейного обхода
+Прелесть среза заключается в его способности выполнять линейные обходы, которые
+механически симпатичны при использовании семантики значений для минимизации выделения
+памяти в куче.
```
Comment (rumyantseva) [inline _content/tour/rus/slices.article:455]:
```suggestion
памяти на куче.
```

You used "на куче" and "на стеке" above, and I think, actually, it is used more often than "в куче" :)

File: _content/tour/rus/slices.article [hunk 10]
```diff
@@ -468,46 +463,46 @@ minimize heap allocations.
...
+не создаются.
 
-All the data in this code stays on the stack. No extra copies of the data inside
-the byte slice are copied.
+** Примечания
 
-** Notes
+- Срезы похожи на динамические массивы с особым встроенным функционалом.
+- Есть разница между длиной и ёмкостью среза, и каждый из них выполняет свою функцию.
+- Срезы позволяют создавать несколько "видов" одного базового массива.
```
Comment (rumyantseva) [inline _content/tour/rus/slices.article:478]:
```suggestion
- Срезы позволяют создавать несколько "представлений" одного базового массива.
```

**Review by rumyantseva — COMMENTED** [review]
Added some suggestions.

File: _content/tour/rus/slices.article [hunk 11]
```diff
@@ -1,29 +1,29 @@
-Slices
-Slices are an incredibly important data structure in Go. They form the basis for how we manage and manipulate data in a flexible, performant, and dynamic way. It's very beneficial for all Go programmers to learn how slices work, and how to use them.
+Срезы
```
[the remark this answers is not in the document]
Comment (koleaby4) [reply _content/tour/rus/slices.article:1]: it is there 👍🏼

File: _content/tour/rus/slices.article [hunk 12]
```diff
@@ -102,17 +103,17 @@ to move the data around my program. This includes declaring fields on a type.
         Z []bool
     }
 
-This goes for all of Go’s internal data structures (slices, maps, channels, interfaces, and functions).
+Это относится ко всем внутренним структурам данных Go (срезам, картам, каналам, интерфейсам и функциям).
```
[the remark this answers is not in the document]
Comment (koleaby4) [reply _content/tour/rus/slices.article:106]: we went with "карта" in the  `maps` article - let's keep it here for consistency

**Comment by koleaby4** [comment]
> Added some suggestions.

very good recommendations, Elena - thank you.
Applied all but 1 -> keeping `карта` as the main term for `map`

**Review by rumyantseva — DISMISSED** [review]

**Review by rumyantseva — APPROVED** [review]

**Pull request merged**